pipenv/utils/safe_ast.py

```diff
@@ -0,0 +1,148 @@
+import ast
```

Comment:
Can you please add a short comment describing the purpose of the new module?

Reply:
Its not really that new, its just an improvement/simplification over what requirementslib did before in terms of ast parsing:  https://github.com/sarugaku/requirementslib/blob/main/src/requirementslib/models/setup_info.py#L442-L777

If you recall:  https://github.com/pypa/pipenv/pull/5357

Basically setup.py has unsafe code that should not be evaluated during resolution (hence why the community has been moving towards pyproject.toml and setup.cfg).